pkg/vulnsrc/openeuler/openeuler.go

```diff
@@ -0,0 +1,289 @@
+				uniqOSVers[pkg.OSVer] = struct{}{}
+				if err := vs.dbc.PutDataSource(tx, pkg.OSVer, source); err != nil {
+					return xerrors.Errorf("failed to put data source: %w", err)
+				}
+			}
+
+			if err := vs.dbc.PutAdvisoryDetail(tx, cvrf.Tracking.ID, pkg.Name,
+				[]string{pkg.OSVer}, advisory); err != nil {
+				return xerrors.Errorf("unable to save %s CVRF: %w", pkg.OSVer, err)
+			}
+		}
+
+		var references []string
+		for _, ref := range cvrf.References {
+			references = append(references, ref.URL)
+		}
+
+		severity := types.SeverityUnknown
+		for _, cvuln := range cvrf.Vulnerabilities {
+			for _, threat := range cvuln.Threats {
+				sev := severityFromThreat(threat.Severity)
+				if severity < sev {
+					severity = sev
+				}
+			}
+		}
+
+		vuln := types.VulnerabilityDetail{
+			References:  references,
+			Title:       cvrf.Title,
+			Description: getDetail(cvrf.Notes),
+			Severity:    severity,
+		}
+		if err := vs.dbc.PutVulnerabilityDetail(tx, cvrf.Tracking.ID, source.ID, vuln); err != nil {
+			return xerrors.Errorf("failed to save openEuler CVRF vulnerability: %w", err)
+		}
+
+		// for optimization
+		if err := vs.dbc.PutVulnerabilityID(tx, cvrf.Tracking.ID); err != nil {
+			return xerrors.Errorf("failed to save the vulnerability ID: %w", err)
+		}
+	}
+	return nil
+}
+
+func getAffectedPackages(productTree ProductTree) []Package {
+	var pkgs []Package
+	var osArches = make(map[string][]string) // OS version => arches
+	for _, branch := range productTree.Branches {
+		// `src` pkgs are the really affected pkgs.
+		if branch.Type != "Package Arch" || branch.Name == "" {
+			continue
+		}
+		for _, production := range branch.Productions {
+			osVer := getOSVersion(production.CPE)
+			if osVer == "" {
+				log.Printf("Unable to parse OS version: %s", production.CPE)
+				continue
+			}
+
+			// Store possible architectures for OS version.
+			// We need this to find affected architectures for src pkg later.
+			if branch.Name != "src" {
+				if arches, ok := osArches[osVer]; ok {
+					osArches[osVer] = append(arches, branch.Name)
+				} else {
+					osArches[osVer] = []string{branch.Name}
+				}
+				continue
+			}
+
+			// e.g., `ignition-2.14.0-2` or `ignition-2.14.0-2.oe2203sp2.src.rpm`
+			pkgName, pkgVersion := parseProduction(production)
+			if pkgName == "" || pkgVersion == "" {
+				log.Printf("Unable to parse Production: %s", production)
+				continue
+			}
+			pkg := Package{
+				Name:         pkgName,
+				FixedVersion: pkgVersion,
+				OSVer:        osVer,
+			}
+			pkgs = append(pkgs, pkg)
+		}
+	}
+
+	// Fill affected architectures
+	for i, pkg := range pkgs {
+		arches := lo.Uniq(osArches[pkg.OSVer])
+		sort.Strings(arches)
+		pkgs[i].Arches = arches
+	}
+
+	return pkgs
+}
+
+func getOSVersion(cpe string) string {
+	// e.g., cpe:/a:openEuler:openEuler:22.03-LTS-SP3
+	parts := strings.Split(cpe, ":")
+	if len(parts) != 5 || parts[2] != "openEuler" {
+		// e.g., cpe:/a:openEuler:openEuler-22.03-LTS
```

Comment:
@DmitriyLewen I did some changes here, to get OS version from the wrong format, please check.

Reply:
@DmitriyLewen Please re-run the tests and tell me what to do next to push this PR merged. Thank you very much!

Reply:
I try not to use `regex` if possible.
I updated the logic -https://github.com/aquasecurity/trivy-db/pull/397/commits/1dbede275804173db158f0143904e0c6bbb382eb
@wjunLu please take a look.

Reply:
Thanks!

Your commit is better.